extra/server-discovery/pkg/es/reindex/reindex.go

```diff
@@ -744,3 +782,129 @@ func (ri *reIndexer) Watch(ctx context.Context) {
 		}
 	}()
 }
+
+func (ri *reIndexer) processResource(source *bytes.Buffer) error {
+	var data map[string]interface{}
+	if err := json.Unmarshal(source.Bytes(), &data); err != nil {
+		return fmt.Errorf("Error unmarshaling: %v\n", err)
+	}
+
+	resourceType, ok := data["resourceType"].(string)
+	if !ok {
+		return fmt.Errorf("resourceType not found or not a string")
+	}
+
+	// only generate embeddings for compose:record for now
+	if resourceType == "compose:record" {
+		catchAll, ok := data["catch_all"].([]interface{})
+		if !ok {
+			return fmt.Errorf("records catch_all not found or not an array")
+		}
+
+		embeddings, err := ri.embedder.GenerateEmbeddings(convertValuesToString(catchAll))
+		if err != nil {
+			return err
+		}
+
+		data["vectorsValue"] = embeddings
+
+		updatedJSON, err := json.Marshal(data)
+		if err != nil {
+			return fmt.Errorf("Error marshaling: %v\n", err)
+		}
+
+		source.Reset()
+		source.Write(updatedJSON)
```

Comment:
I'd prefer you return an updated stream over updating the one passed in arguments